bootloader/src/pyi_pythonlib.c

```diff
@@ -88,12 +88,13 @@ pyi_pylib_load(ARCHIVE_STATUS *status)
       strncpy(dllname, status->cookie.pylibname, 64);
     }
 #else
-    len = 0;
     strncpy(dllname, status->cookie.pylibname, 64);
 #endif
 
     if (len >= 64 || dllname[64-1] != '\0') {
-        FATALERROR("DLL name length exceeds buffer\n");
+        dllname[64-1] = 0;
+        FATALERROR("DLL name (%s) length (%d) exceeds dllname[64]\n",
+                   dllname, (len) ? len : strlen(status->cookie.pylibname));
```

Comment:
```suggestion
        FATALERROR("DLL name (%s) length exceeds buffer (64 chars)\n",
                   status->cookie.pylibname);
```
since `dllname` will  probably be cut off. 

No need to print the length, since it is on low interest - and if one could count it.

Or did I miss something?

Reply:
as to printing the len value. - easy to give, and it may help with documenting an issue.

As the fixed size of the array - Actually I am puzzled why it is limited to 64 chars. IIRC AIX supports filenames up to 256 characters. Maybe that is PATH length. I do know that dlopen() searches LIBPATH if there is no "/" in the name (aka relative path). When relative or exact pathname only that one argument is searched.

Reply:
Re 64 chars: This is not a PATH, but only a well-known filename, which is either build from `"libpython%01d.%01d.a(libpython%01d.%01d.so)"` (about 40 chars) or from `status->cookie.pylibname` (limited to 64 chars).

Beside there is some serous error (`status->cookie.pylibname` is not properly terminated), this error should never ever be triggered.

Reply:
I found it useful to see which part of the test had failed - which is why I added printing the value of `len`. And I like seeing that the object name referenced is what is expected.

Hope this works for you as well.